### PR TITLE
Support count in orderby and filter query options

### PR DESCRIPTION
**Pull request by genusP**
### Issues

fix issue #194 

### Description

This PR based on #738 

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: src/Microsoft.AspNet.OData.Shared/Query/Expressions/FilterBinder.cs [hunk 1]
```diff
@@ -1425,6 +1462,12 @@ private Expression BindCustomMethodExpressionOrNull(SingleValueFunctionCallNode
         /// <returns>The LINQ <see cref="Expression"/> created.</returns>
         private Expression BindSingleValueNode(SingleValueNode node)
         {
+            CountNode countNode = node as CountNode;
+            if (countNode != null)
+            {
+                return BindCountNode(countNode);
+            }
+
             switch (node.Kind)
```
Comment (mikepizzo) [inline src/Microsoft.AspNet.OData.Shared/Query/Expressions/FilterBinder.cs]: Any reason this is not in the switch statement for case: QueryNodeKind.Count?

File: src/Microsoft.AspNet.OData.Shared/Query/OrderByNode.cs [hunk 2]
```diff
@@ -60,6 +60,11 @@ public static IList<OrderByNode> CreateCollection(OrderByClause orderByClause)
             List<OrderByNode> result = new List<OrderByNode>();
             for (OrderByClause clause = orderByClause; clause != null; clause = clause.ThenBy)
             {
+                if (clause.Expression is CountNode)
+                {
+                    result.Add(new OrderByCountNode(clause));
+                    continue;
+                }
                 if (clause.Expression is NonResourceRangeVariableReferenceNode ||
```
Comment (mikepizzo) [inline src/Microsoft.AspNet.OData.Shared/Query/OrderByNode.cs]: According to style guidelines, should insert a blank line after closing brace.

**Comment by mikepizzo** [comment]
@genusP - Thanks for the contribution! This should make it into our upcoming 7.2 release.

File: src/Microsoft.AspNet.OData.Shared/Query/Expressions/FilterBinder.cs (hunk 1 above)
Comment (genusP) [reply src/Microsoft.AspNet.OData.Shared/Query/Expressions/FilterBinder.cs]: No reason. Fixed.

File: test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Builder/FunctionConfigurationTest.cs [hunk 3]
```diff
@@ -830,6 +830,27 @@ public void GetEdmModel_SetsDateTimeAsParameterType_WorksForDefaultConverter()
             Assert.True(parameter.Type.IsNullable);
         }
 
+        [Fact]
+        public void GetEdmModel_SetDecimalAsParameterType()
```
Comment (KanishManuja-MS) [inline test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Builder/FunctionConfigurationTest.cs]: >GetEdmModel_SetDecimalAsParameterType [](start = 20, length = 37)

I wanted to make sure that I am not missing something here. Is this relevant to the change?

Comment (genusP) [reply test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Builder/FunctionConfigurationTest.cs]: You are right. I remove this changes.
